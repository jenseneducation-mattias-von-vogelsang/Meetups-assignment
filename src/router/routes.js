//******************************/
//*** ROUTES FOR VUE ROUTER ***/
//****************************/
const routes = [
  {
    path: '/',
    name: 'Home',
    component: () => import('../views/Home/Home.vue'),
  },
  {
    path: '/Account',
    name: 'Account',
    component: () => import('../views/Account/Account.vue'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/CreateMeetup',
    name: 'CreateMeetup',
    component: () => import('../views/CreateMeetup/CreateMeetup.vue'),
    meta: {
      requiresAuth: true,
    },
  },
  {
    path: '/Login',
    name: 'Login',
    component: () => import('../views/Login/Login.vue'),
    meta: {
      requiresGuest: true,
    },
  },
  {
    path: '/Register',
    name: 'Register',
    component: () => import('../views/Register/Register.vue'),
    meta: {
      requiresGuest: true,
    },
  },
  {
    path: '/MeetupDetails/:id',
    name: 'MeetupDetails',
    component: () => import('../views/MeetupDetails/MeetupDetails.vue'),
  },
];

export default routes;
