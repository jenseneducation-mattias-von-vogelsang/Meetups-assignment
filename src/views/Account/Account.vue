<template>
  <div>
    <UserPage />
  </div>
</template>
<script>
import UserPage from '@/components/UserPage/UserPage.vue';
export default {
  name: 'Account',
  components: {
    UserPage,
  },
};
</script>
