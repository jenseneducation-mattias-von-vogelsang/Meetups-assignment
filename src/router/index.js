import Vue from 'vue';
import VueRouter from 'vue-router';
import routes from './routes.js';
// import store from '../store/index.js';

Vue.use(VueRouter);

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
});

//************************************/
//*** NAVIGATION GUARDS FOR LOGIN ***/
//**********************************/

//Router navigation guards for login & account page

// router.beforeEach((to, from, next) => {
//   if (to.matched.some((record) => record.meta.requiresAuth)) {
//     if (!store.getters.isLoggedIn) {
//       // Redirect to the Login Page
//       next('/Login');
//     } else {
//       next();
//     }
//   } else if (to.matched.some((record) => record.meta.requiresGuest)) {
//     if (store.getters.isLoggedIn) {
//       // Redirect to the Login Page
//       next('/Account');
//     } else {
//       next();
//     }
//   } else {
//     next();
//   }
// });

export default router;
