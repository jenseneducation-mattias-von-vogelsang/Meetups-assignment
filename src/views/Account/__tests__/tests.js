import {
  createLocalVue,
  shallowMount,
  mount,
  enableAutoDestroy,
} from '@vue/test-utils';
import Vuex from 'vuex';
import Account from '@/views/Account/Account.vue';
import VueRouter from 'vue-router';
import router from '@/router/index.js';
import Vue from 'vue';
import Vuetify from 'vuetify';

const localVue = createLocalVue();
localVue.use(VueRouter, Vuex);

//Destroy wrapper after each test
enableAutoDestroy(afterEach);

describe('User navigates to account page', () => {
  let vuetify, getters, actions, state;

  beforeEach(() => {
    vuetify = new Vuetify();
    Vue.use(Vuetify);

    state = {
      user: {
        _id: '123123123123123123123',
        createdMeetups: [
          {
            _id: '123123123',
            location: {
              coordinates: [123, 123],
              type: 'Point',
            },
            attendees: ['Mattias'],
            reviews: [],
            eventName: 'Test',
            host: 'Test',
            details: 'Testing',
            time: '12:34',
            date: '2020-02-02',
            address: 'Test street 2',
          },
        ],
        attendingMeetups: [
          {
            _id: '123123123',
            eventName: 'Test',
          },
        ],
        reviewHistory: [
          {
            _id: '123123123',
            eventName: 'Test',
            review: 'Testing this test',
          },
        ],
        name: 'Testington',
        username: 'Testzor',
        password: 'NotThisTime!',
        email: 'Testboi@testing.test',
        date: '2020-09-27T00:04:53.496+00:00',
      },
    };

    getters = {
      user: () => state.user,
    };

    actions = {
      getProfile: jest.fn(),
      viewFullMeetup: jest.fn(),
      removeAttendThisMeetup: jest.fn(),
      removeThisReview: jest.fn(),
      removeThisMeetup: jest.fn(),
    };
  });
  test('It renders correctly and matches snapshot', async () => {
    const store = new Vuex.Store({
      actions,
      state,
      getters,
    });
    const wrapper = mount(Account, {
      store,
      localVue,
      router,
      vuetify,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
