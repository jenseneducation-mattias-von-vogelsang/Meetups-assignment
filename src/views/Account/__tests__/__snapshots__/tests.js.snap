// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User navigates to account page It renders correctly and matches snapshot 1`] = `
<div>
  <div class="container fill-height container--fluid">
    <div class="mx-auto v-card v-sheet theme--light rounded-0" style="min-height: 100%; max-width: 800px;">
      <div class="v-image v-responsive theme--light" style="max-height: 300px;">
        <transition-stub name="fade-transition" mode="in-out">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        </transition-stub>
        <transition-stub name="fade-transition" appear="true">
          <div class="v-image__placeholder">
            <div class="px-3 pt-3 pb-3 fill-height v-sheet theme--light grey lighten-4">
              <div aria-busy="true" aria-live="polite" role="alert" class="v-skeleton-loader mx-auto v-skeleton-loader--is-loading theme--light">
                <div class="v-skeleton-loader__image v-skeleton-loader__bone"></div>
              </div>
            </div>
          </div>
        </transition-stub>
        <div class="v-responsive__content"></div>
      </div>
      <div class="col">
        <div class="v-avatar avatar-position" style="height: 150px; min-width: 150px; width: 150px;">
          <div class="v-image v-responsive theme--light">
            <transition-stub name="fade-transition" mode="in-out">
              <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
            </transition-stub>
            <div class="v-responsive__content"></div>
          </div>
        </div>
      </div>
      <div tabindex="-1" class="margin-top v-list-item theme--light" style="color: rgba(0, 0, 0, 0.4); caret-color: rgba(0, 0, 0, .4);">
        <div class="v-list-item__content">
          <div class="v-list-item__title display-1">Testington</div>
          <div class="v-list-item__subtitle title">Testboi@testing.test</div>
          <div class="v-list-item__subtitle title">Testzor</div>
          <div class="v-list-item__subtitle title">Member since: 2020-09-27T00:04:53.496+00:00</div>
        </div>
      </div>
      <div data-test="createdMeetups" class="container full">
        <div class="v-card__title">My created Meetups
          <span class="v-badge theme--light"><span class="v-badge__wrapper"><transition-stub name="scale-rotate-transition"><span aria-atomic="true" aria-label="Badge" aria-live="polite" role="status" class="v-badge__badge orange" style="bottom: calc(100% - -2px); left: calc(100% - 1px);">1</span></transition-stub></span></span>
        </div>
        <div class="row">
          <div xs="8" class="col-md-4 col">
            <div class="mx-auto d-flex flex-column align-center v-card v-sheet theme--light" style="min-height: 50px; min-width: 200px;">
              <div class="v-card__title clickable">Test</div> <button type="button" class="v-btn v-btn--contained theme--dark v-size--default orange" style="width: 100%;" id="removeMeetup"><span class="v-btn__content">
              Remove Meetup
            </span></button>
            </div>
          </div>
        </div>
      </div>
      <div data-test="attendingMeetups" class="container">
        <div class="v-card__title">Meetups I'm attending
          <span class="v-badge theme--light"><span class="v-badge__wrapper"><transition-stub name="scale-rotate-transition"><span aria-atomic="true" aria-label="Badge" aria-live="polite" role="status" class="v-badge__badge orange" style="bottom: calc(100% - -2px); left: calc(100% - 1px);">1</span></transition-stub></span></span>
        </div>
        <div class="row">
          <div xs="8" class="col-md-4 col">
            <div class="mx-auto d-flex flex-column align-center v-card v-sheet theme--light" style="min-height: 50px; min-width: 200px;">
              <div class="v-card__title clickable"></div> <button type="button" class="v-btn v-btn--contained theme--dark v-size--default orange" style="width: 100%;" id="removeAttendance"><span class="v-btn__content">
              Remove attendance
            </span></button>
            </div>
          </div>
        </div>
      </div>
      <div data-test="reviewHistory" class="container">
        <div class="v-card__title">Meetups I've reviewed
          <span class="v-badge theme--light"><span class="v-badge__wrapper"><transition-stub name="scale-rotate-transition"><span aria-atomic="true" aria-label="Badge" aria-live="polite" role="status" class="v-badge__badge orange" style="bottom: calc(100% - -2px); left: calc(100% - 1px);">1</span></transition-stub></span></span>
        </div>
        <div class="row">
          <div xs="8" class="col-md-4 col">
            <div class="mx-auto d-flex flex-column align-center v-card v-sheet theme--light" style="min-height: 50px; min-width: 200px;">
              <div class="v-card__title clickable"></div>
              <div class="v-card__subtitle">
                Testing this test
              </div> <button type="button" class="v-btn v-btn--contained theme--dark v-size--default orange" style="width: 100%;" id="removeReview"><span class="v-btn__content">
              Remove review
            </span></button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
